tests/unit/TrieTest.js

```diff
@@ -0,0 +1,43 @@
+import Trie from '../../src/libs/Trie';
+
+describe('Trie', () => {
+    it('Test if a node can be found in the Trie', () => {
+        const emojisTrie = new Trie();
+        emojisTrie.add('grinning', {code: '😀'});
+        emojisTrie.add('grin', {code: '😁'});
+        emojisTrie.add('joy', {code: '😂'});
+        emojisTrie.add('rofl', {code: '🤣'});
+        expect(emojisTrie.isWord('eyes')).toBeNull();
+        expect(emojisTrie.isWord('joy').metaData).toEqual({code: '😂'});
+    });
+
+    it('Test finding all leaf nodes starting with a substring', () => {
+        const wordTrie = new Trie();
+        wordTrie.add('John');
+        wordTrie.add('Robert');
+        wordTrie.add('Robertson');
+        wordTrie.add('Rock');
+        const expected = ['Robert', 'Robertson', 'Rock'];
+        expect(wordTrie.getAllMatchingWords('Ro').sort()).toEqual(expected.sort());
+    });
+
+    it('Test finding only the first 5 matching emojis', () => {
```

Comment:
because `Github` picker only has 5 emojis, we don't need to continue searching the tree after that.

![Screen Shot 2022-08-30 at 1 00 18 PM](https://user-images.githubusercontent.com/108357004/187421602-9bc7b4e5-c74a-415c-97dc-bebb613211da.png)